utils/blog_helpers.ts

```diff
@@ -7,25 +7,44 @@ export function orderPosts(posts) {
   return posts.slice().sort(sortByDate)
 }
 
-const detectShortcodes = /\{\{(.*?)\}\}/gm
-const preStrip = content => {
-  return content.replace(detectShortcodes, '')
-}
+export function stripMarkdown(content: string): string {
+  content = content.replace(/\{\{(.*?)\}\}/gm, '');
+
+  try {
+    content = content.replace(/<[^>]*>/g, ''); // Remove all elements that look like <tags>
+    content = content.replace(/^([\s\t]*)([\*\-\+]|\d+\.)\s+/gm, ''); // Strip out bullet points or numbered lists
+
+    content = content
+      .replace(/~~/g, '')
+      .replace(/`{3}.*\n([\s\S]*?)\n`{3}/g, '$1')
+      .replace(/~{3}.*\n/g, '');
+
+    content = content
+      .replace(/\[([^\]]+)\]\([^\)]+\)/g, '$1')
+      .replace(/!\[([^\]]*)\]\([^\)]+\)/g, '$1')
+      .replace(/#{1,6}\s*(.*)/g, '$1')
+      .replace(/>\s*(.*)/g, '$1')
+      .replace(/(\*\*|__)(.*?)\1/g, '$2')
+      .replace(/(\*|_)(.*?)\1/g, '$2')
+      .replace(/`([^`]+)`/g, '$1')
+      .replace(/^(\n)?\s{0,}#{1,6}\s*( (.+))? +#+$|^(\n)?\s{0,}#{1,6}\s*( (.+))?$/gm, '$1$3$4$6')
+      .replace(/~(.*?)~/g, '$1');
+
+    content = content.replace(/<[^>]*>/g, '');
```

Comment:
AI thinks this is alright.

🤖
The code is not "weird," but it is quite aggressive in sanitizing and formatting text. It aims to strip out various types of formatting, tags, and lists to convert the content to plain text.
🤖

Reply:
FYI, we checked out a few packages that achieve the same results and they use very similar code although some more comments could be useful @joshbermanssw  😉

Reply:
Also the tests help with the code being a bit scary.

Reply:
added some more comments! Did you give AI the context that the method is used as a markdownStripper?